tapa/verilog/xilinx/module.py

```diff
@@ -831,6 +833,28 @@ def get_streams_fifos(module: Module, streams_name: str) -> list[str]:
             number = match.group(1)
             fifos.add(f"{streams_name}_{number}")
 
+    # singleton array without number
+    # when a stream array has length 1,
+    # the generated port name may not have the number
+    # in it.
+    # e.g., in tests/functional/singleton/vadd.cpp,
+    # "tapa::istreams<float, M>& a" where M = 1 resulted in
+    # stream data port a_s_dout rather than a_0_dout.
+    if not fifos:
+        for s in module.ports:
+            for infix in FIFO_INFIXES:
+                if s.startswith(f"{streams_name}{infix}"):
+                    if fifos:
```

Comment:
This wouldn't be possible since we are under the `if not fifos` branch. Did you want to remove the `if not fifos:` conditional?